pkg/gather/collect/collect.go

```diff
@@ -472,10 +532,6 @@ func (c *Collector) CollectObject(ctx context.Context, u *unstructured.Unstructu
 func (c *Collector) CollectResource(ctx context.Context, resourceInfo *ResourceInfo, namespace, name string) error {
 	obj, err := c.dynamicClient.Resource(resourceInfo.Resource).Namespace(namespace).Get(ctx, name, metav1.GetOptions{})
 	if err != nil {
-		if errors.IsNotFound(err) {
```

Comment:
Yeah, essentially both options suck. This can either silently skip wrongly specified resource or have issue collecting clusters that have creation/deletion hot-loops on some resources. Putting that aside, in a working cluster the issue is only with ephemeral resources that may force you to run this again.

Let me see if I can come up with something smarter.

Reply:
I've recalled. This is inside the collector, so this should get an error and the caller can decide. Some tools can translate it into a warning, some fail, and test should fail with it. See
https://github.com/scylladb/scylla-operator/blob/a6cb4e692855e2073212faa097917d0cef8b0c64/pkg/cmd/operator/mustgather.go#L315-L322